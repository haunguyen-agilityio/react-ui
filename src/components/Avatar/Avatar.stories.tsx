// import { Story } from '@storybook/react'

// // components
// import Avatar, { AvatarProp } from '@components/Avatar'

// // styles
// import '@components/Avatar/styles'

// export default {
//   title: 'Components/Avatar',
//   component: Avatar,
// }

// const Template: Story<AvatarProp> = (args) => <Avatar {...args} />

// export const AvatarDefault = Template.bind({})
// AvatarDefault.args = {
//   image: './src/avatar.jpg',
//   firstName: 'Jame',
//   sizeAvatar: 50,
//   variant: 'circular',
//   isEdit: false,
// }

// export const AvatarEdit = Template.bind({})
// AvatarEdit.args = {
//   image: './src/avatar.jpg',
//   firstName: 'Jame',
//   sizeAvatar: 50,
//   variant: 'circular',
//   isEdit: true,
// }

// export const AvatarImage = Template.bind({})
// AvatarImage.args = {
//   image: 'https://cdn2.vectorstock.com/i/1000x1000/70/51/worried-man-avata-vector-19437051.jpg',
//   firstName: 'Jame',
//   sizeAvatar: 50,
//   variant: 'circular',
//   isEdit: true,
// }

// export const AvatarSmall = Template.bind({})
// AvatarSmall.args = {
//   image: './src/avatar.jpg',
//   firstName: 'Jame',
//   sizeAvatar: 30,
//   variant: 'circular',
//   isEdit: true,
// }

export {}
