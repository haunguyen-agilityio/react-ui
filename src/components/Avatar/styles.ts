// lib MUI
import { Avatar } from '@mui/material'
import { styled } from '@mui/system'

// theme
import theme from '../../themes'

export const StyledAvatar = styled(Avatar)({
  backgroundColor: '#21d083',
  fontWeight: 600,
  color: theme.palette.common.white,
})

export const SmallAvatar = styled(Avatar)({
  backgroundColor: '#aaaaaa',
})
