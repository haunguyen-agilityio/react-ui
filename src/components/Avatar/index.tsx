import { memo } from 'react'

// lib MUI
import { Badge } from '@mui/material'

// styles
import { StyledAvatar, SmallAvatar } from './styles'

// icon
import EditIcon from '@mui/icons-material/Edit'

export type AvatarProp = {
  image: string
  firstName: string
  onClick?: () => void
  sizeAvatar: number
  variant: 'circular' | 'rounded' | 'square'
  isEdit: boolean
}

const Avatar = ({
  image = './src/image.jpg',
  firstName = 'Jame',
  onClick = () => {},
  sizeAvatar = 50,
  variant = 'circular',
  isEdit = false,
}: AvatarProp) => {
  return (
    <div onClick={onClick}>
      {isEdit && (
        <Badge
          sx={{ position: 'relative', top: sizeAvatar * 0.25, left: sizeAvatar * 0.86 }}
          overlap='circular'
          badgeContent={
            <SmallAvatar sx={{ width: sizeAvatar / 2.2, height: sizeAvatar / 2.2 }} alt='Edit'>
              <EditIcon sx={{ fontSize: sizeAvatar * 0.3, color: '#fff' }} />
            </SmallAvatar>
          }
        ></Badge>
      )}
      <StyledAvatar
        variant={variant}
        sx={{ width: sizeAvatar, height: sizeAvatar, fontSize: sizeAvatar * 0.62 }}
        alt={firstName}
        src={image}
      />
    </div>
  )
}

export default memo(Avatar)
