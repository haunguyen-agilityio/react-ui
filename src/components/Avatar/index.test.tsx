// component
import Avatar, { AvatarProp } from '../Avatar'

// lib
import { fireEvent, render, screen } from '@testing-library/react'

describe('Test Avatar', () => {
  const defaultProps: AvatarProp = {
    image: './src/image.jpg',
    firstName: 'Jame',
    onClick: jest.fn(),
    sizeAvatar: 50,
    variant: 'circular',
    isEdit: false,
  }

  const customProps: AvatarProp = {
    image: './src/image.jpg',
    firstName: 'Jame',
    onClick: jest.fn(),
    sizeAvatar: 50,
    variant: 'circular',
    isEdit: true,
  }

  test('It should match DOM Snapshot', () => {
    const { container } = render(<Avatar {...defaultProps} />)

    expect(container).toMatchSnapshot()
  })

  test('It should render correctly', () => {
    render(<Avatar {...defaultProps} />)

    expect(screen.getByRole('img')).toBeInTheDocument()
  })

  test('It should call onClick Avatar', () => {
    render(<Avatar {...defaultProps} />)
    fireEvent.click(screen.getByRole('img'))

    expect(defaultProps.onClick).toHaveBeenCalled()
    expect(defaultProps.onClick).toHaveBeenCalledTimes(1)
  })

  test('It should call onClick Avatar Edit', () => {
    render(<Avatar {...customProps} />)
    fireEvent.click(screen.getByRole('img'))

    expect(customProps.onClick).toHaveBeenCalled()
    expect(customProps.onClick).toHaveBeenCalledTimes(1)
  })
})
