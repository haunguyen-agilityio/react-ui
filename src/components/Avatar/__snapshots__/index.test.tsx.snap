// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Avatar It should match DOM Snapshot 1`] = `
<div>
  <div>
    <div
      class="MuiAvatar-root MuiAvatar-circular css-1cmlm5z-MuiAvatar-root"
    >
      <img
        alt="Jame"
        class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
        src="./src/image.jpg"
      />
    </div>
  </div>
</div>
`;
